### ISSUE TITLE
Many games cannot be launched if I use Vulkan

### ISSUE THREAD

**Issue by Skiski** [ValveSoftware/Proton#6262]
Today, I don't know why, many games are crashing during launch. Ir is the case for Jedi Fallen Order, NFS Most Wanted, Burnout Paradise Remastered, Brawhalla, Dredge Chapter One, Rayman Origins and Trials Evolution.

I can only tell that Jedi Fallen Order was working great last week. I haven't launch the others since a long time, or never at all, but they should be working according to the reports.

What is strange is that if I add `PROTON_USE_WINED3D=1`, they all launch, excepted Burnout Paradise Remastered. Of course, the performances are not very good.

On the other hand, games like Forza Horizon 4 are running great with Vulkan.

I haven't updated my system since last week. I made an update since then, and I upgraded from nvidia 515 drivers to 520, but it did not change anything.

I'm on Fedora 36, with a Nvidia rtx 3070 ti.

**Comment by kisak-valve** [comment]
Hello @Skiski, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. Also, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to a couple game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

**Comment by Skiski** [comment]
My system infos:
https://gist.github.com/Skiski/f4cd8a624d8a38a7599f37b650335dcf
And I've attached the log file for Jedi Fallen Order.
[steam-1172380.log](https://github.com/ValveSoftware/Proton/files/9845400/steam-1172380.log)
I hope this can help.

---

I've tried other games. Far Cry 3 is crashing but Horizon Zero Dawn (on max settings), Black Future '88, Just Cause 3, Soul Calibur VI and REZ Infinite are working.

EA and Ubisoft games seem to be affected. It may be related to the fact that they are using a launcher. It crashes before the launcher can be displayed. But I don't see how it can be related to Vulkan.

**Comment by kisak-valve** [comment]
Unfortunately, you've copied the system information from Steam before it was able to finish collecting the extended diagnostics information.

Are you using Steam's client beta, and if you are, does opting out of it have any effect on this issue?

**Comment by mattc927** [comment]
I'm experiencing the same. Some games that used to work in Proton with DXVK now don't launch at all unless set to use WINED3D.
System info: https://gist.github.com/mattc927/553a3a7ed452826acb26681aae05c416
Log for trying to run Black Ops II Zombies (212910)
[steam-212910.log](https://github.com/ValveSoftware/Proton/files/9845517/steam-212910.log)


Interestingly, opting **out** of the beta did get BOII to launch again, so it does seem to be related to that.

**Comment by spiffeeroo** [comment]
Try opting out of the Steam Beta Client (Oct 22, 2022). DXVK games may have issues with the newest Steam Beta. Native Linux OpenGL games, DX8 games, and PROTON_USE_WINED3D=1 games work with Steam Beta, however, with the limited games I tested.

Edit: I tested Hollow Knight (native Vulkan game) on Steam Beta and it works. 

**Steam Client Beta and Proton 7.0-4:**

- Hollow Knight appid 367520 (native Vulkan) = works
- Cities Skylines appid 255710, Baba Is You appid 736260 (native OpenGL) = works
- Persona 4 Golden appid 1113000 (DXVK) = crashes on startup
- Persona 4 Golden appid 1113000 (PROTON_USE_WINED3D=1) = works
- Ori and the Blind Forest: Definitive Ediiton appid 387290 (DXVK) = crashes on startup
- Ori and the Blind Forest: Definitive Edition appid 387290 (PROTON_USE_WINED3D=1) = works

Steam Client (stable) has all these games working.

**Comment by kisak-valve** [comment]
I'm seeing vkbasalt in common between the logs. Can one of you test temporarily disabling mangohud and/or vkbasalt and see if that has any effect on the game's behavior with the Steam client beta?

**Comment by mattc927** [comment]
Removed vkbasalt, mangohud, and goverlay from my system and ran beta again. Still crashed.
Updated system info: https://gist.github.com/mattc927/50d1982bd947d7f421465311bbf1c410 (fixed link)
[steam-212910.log](https://github.com/ValveSoftware/Proton/files/9845556/steam-212910.log)
Updated log

**Comment by plasmamax1** [comment]
Just here to add that certain games using Proton do not launch at all. It Takes Two and Batman: Arkham City GOTY do not launch, while Yakuza 0 has no issues launching. There are no command line arguments added to any game.

I did see I was on the steam beta channel. Switching back to the stable version fix the issue as all games are able to launch as normal. I am running steam-runtime on archlinux with linux-zen-6.0.2-1 and mesa-git.

**Comment by Skiski** [comment]
Indeed, opting out of the Steam Beta solved the problem. I hope that you will be able to find what is causing this in the beta.
Thanks for your quick answer.

---

I've updated my system informations, but without the steam beta.

**Comment by Frogging101** [comment]
It is a problem with the `ubuntu12_32/libVkLayer_steam_fossilize.so` shipped in the Beta. A workaround is to disable shader pre-caching in Steam settings (note: I think this causes redownloading of fossilize/pre-caching data when re-enabled).
Other affected games:
- Devil May Cry 4
- Fallout 3 GOTY
- Just extrapolating from the pattern: Probably any other game that's 32-bit and uses DXVK

**Comment by ghost** [comment]
Adding to the List above: 
Warhammer 3 is also broken with the Steam Beta - Wine Preloader crashing on the Launcher.
Opting out of Steam Beta instantly fixed this.

Fedora 36, Wayland, Nvidia 520.56.06, Flatpak Steam

**Comment by spiffeeroo** [comment]
32-bit DXVK games seem to crash with the Steam Beta. 64-bit DXVK games work for me.
Adding to the list:

Works:
- Final Fantasy XII: The Zodiac Age (64-bit) [but the included Config Tool (32-bit) crashes.]
- Bloodstained: Ritual of the Night (64-bit)
- Dark Souls Remastered (64-bit)
- Dark Souls III (64-bit)
- Ori and the Will of the Wisp (64-bit)

Crashes:
- Final Fantasy IX (32-bit)
- Persona 4 Golden (32-bit)
- Sonic Mania (32-bit)
- Shantae and the Pirate's Curse (32-bit)
- Ori and the Blind Forest: Definitive Edition (32-bit)
- Mushihimesama (32-bit)

**Comment by xabdev** [comment]
Since yesterday I'm having problems with Guilty Gear XRD REV2, switching off from steam beta fixed the issue with guilty gear and other dx9 games.

**Comment by StarterX4** [comment]
Can confirm the same issue with the **native** Left 4 Dead 2 Vulkan (OpenGL works fine), and with the latest Steam Client Beta, it fails to run:
```
SDL video target is 'x11'
SDL video target is 'x11'
Using shader api: shaderapivk
free(): invalid pointer
/opt/gry/steamapps/common/Left4Dead2/./hl2.sh: linia 72: 33947 Przerwane               ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
Game process removed: AppID 550 "gamemoderun SDL_AUDIODRIVER=pulseaudio SDL_DYNAMIC_API=/usr/lib32/libSDL2-2.0.so.0 DRI_PRIME=0 LD_PRELOAD=/usr/lib32/libSDL2-2.0.so.0:/usr/lib32/libgcrypt.so.11:/home/starterx4/.steam/steam/ubuntu12_32/gameoverlayrenderer.so LC_ALL=pl_PL.UTF-8 LANG=pl_PL /kesz/Steam/steam/ubuntu12_32/reaper SteamLaunch AppId=550 -- /kesz/Steam/steam/ubuntu12_32/steam-launch-wrapper -- '/opt/gry/steamapps/common/Left4Dead2/./hl2.sh' -game left4dead2 -steam -novid -nojoy -high +exec autoexec.cfg -tickrate 128 -threads 12 +cl_forcepreload 1 +sv_consistency 0 -vulkan", ProcID 33934 
ThreadGetProcessExitCode: no such process 33947
```
After debugging it with `GAME_DEBUGGER=gdb`, it points to the `libdxvk_d3d9.so` library:
```
Thread 1 "hl2_linux" received signal SIGABRT, Aborted.
0xf7fc4509 in __kernel_vsyscall ()
(gdb) where
#0  0xf7fc4509 in __kernel_vsyscall ()
#1  0xf7a983d7 in pthread_kill () from /usr/lib32/libc.so.6
#2  0xf7a399c7 in raise () from /usr/lib32/libc.so.6
#3  0xf7a1e126 in abort () from /usr/lib32/libc.so.6
#4  0xf7a88d64 in ?? () from /usr/lib32/libc.so.6
#5  0xf7aa3bbf in ?? () from /usr/lib32/libc.so.6
#6  0xf7aa5c0f in ?? () from /usr/lib32/libc.so.6
#7  0xf7aa8506 in free () from /usr/lib32/libc.so.6
#8  0xe75d7078 in operator delete(void*) () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#9  0xe75d7108 in operator delete[](void*) () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#10 0xe75dc2df in std::locale::_Impl::_M_install_facet(std::locale::id const*, std::locale::facet const*) ()
   from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#11 0xe75dc710 in std::locale::_Impl::_Impl(unsigned int) () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#12 0xe75dd53e in std::locale::_S_initialize_once() () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#13 0xf7a9b6eb in ?? () from /usr/lib32/libc.so.6
#14 0xe75dd59d in std::locale::_S_initialize() () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#15 0xe75dd5f4 in std::locale::locale() () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#16 0xe75da2b5 in std::ios_base::Init::Init() () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#17 0xe74bb0d2 in ?? () from /opt/gry/steamapps/common/Left4Dead2/bin/libdxvk_d3d9.so
#18 0xf7fcaf9b in ?? () from /lib/ld-linux.so.2
#19 0xf7fcb089 in ?? () from /lib/ld-linux.so.2
#20 0xf7fd1d2b in ?? () from /lib/ld-linux.so.2
#21 0xf7b96724 in _dl_catch_exception () from /usr/lib32/libc.so.6
#22 0xf7fd1dda in ?? () from /lib/ld-linux.so.2
#23 0xf7b966c9 in _dl_catch_exception () from /usr/lib32/libc.so.6
#24 0xf7fd2153 in ?? () from /lib/ld-linux.so.2
#25 0xf7a90ffc in ?? () from /usr/lib32/libc.so.6
#26 0xf7b966c9 in _dl_catch_exception () from /usr/lib32/libc.so.6
#27 0xf7b96784 in _dl_catch_error () from /usr/lib32/libc.so.6
#28 0xf7a90a27 in ?? () from /usr/lib32/libc.so.6
#29 0xf7a910c8 in dlopen () from /usr/lib32/libc.so.6
#30 0xe965906e in ?? () from /opt/gry/steamapps/common/Left4Dead2/bin/materialsystem.so
#31 0xe9659164 in ?? () from /opt/gry/steamapps/common/Left4Dead2/bin/materialsystem.so
#32 0xe96593c5 in ?? () from /opt/gry/steamapps/common/Left4Dead2/bin/materialsystem.so
--Type <RET> for more, q to quit, c to continue without paging--c
#33 0xe95fdbaa in ?? () from /opt/gry/steamapps/common/Left4Dead2/bin/materialsystem.so
#34 0xe95fdc59 in ?? () from /opt/gry/steamapps/common/Left4Dead2/bin/materialsystem.so
#35 0xf613b5c0 in ?? () from bin/launcher.so
#36 0xf61568ee in ?? () from bin/launcher.so
#37 0xf6156a89 in ?? () from bin/launcher.so
#38 0xf6156aa8 in ?? () from bin/launcher.so
#39 0xf613ca7e in LauncherMain () from bin/launcher.so
#40 0x080484c4 in main ()
(gdb) q
```
Opting out of the Client Beta fixed the issue, and now L4D2 VK works fine.

Also, I always have shader precaching disabled.

**Comment by steef435** [comment]
FWIF, Elden Ring also appears to be affected (switched off beta, works fine again).

**Comment by EarlofBurl** [comment]
Civilization V **does not ** work after

• switching from Beta to Stable steam
• disabling shader pre caching and
• trying both DX 11 and 9 and
• multiple Proton Versions including GE.

edit: a little more info. The game worked fine on Friday.
Using Fedora 36 with Wayland + NVidia prop. drivers

**Comment by bcb12** [comment]
Fallout 3 GOTY and Fallout New Vegas do not work after Saturday. Changing to Proton 4.11-13 solves it.

Also Grand Theft Auto IV: The Complete Edition did not work after Saturday as well, and changing the Proton version does not seem to work. Disabling Shader Pre-Caching solved it for me.

**Comment by VladimirMrzv** [comment]
problem with all 32 games/launchers. Solution in post https://github.com/ValveSoftware/Proton/issues/6262#issuecomment-1288136184

But on my system Ubuntu 22.04 I do not have this problem, i think it happens on new systems.

**Comment by deathxxx123** [comment]
All working with Pop!_OS 22.04 + Nvidia + steam stable.

**Comment by kisak-valve** [comment]
Hello, per "Fix a crash launching 32 bit Vulkan titles (including DXVK SteamPlay)" in the [2022-10-24](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3428954061821341725) Steam client beta update, please opt back into the Steam client beta if you opted out to workaround this issue, and/or re-enable Steam's shader pre-caching subcomponent if you disabled that as a workaround and retest this issue.

**Comment by StarterX4** [comment]
> Hello, per "Fix a crash launching 32 bit Vulkan titles (including DXVK SteamPlay)" in the [2022-10-24](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3428954061821341725) Steam client beta update, please opt back into the Steam client beta if you opted out to workaround this issue, and/or re-enable Steam's shader pre-caching subcomponent if you disabled that as a workaround and retest this issue.

L4D2 Vulkan still crashes with the same terminal output.

CachyOS (Arch-based), AMD, mesa 22.2.2, KDE Plasma Wayland.

**Comment by mattc927** [comment]
So far so good with the games I've tested (I've not tried L4D2 yet, though that should be native). Beta 2022-10-24 seems to have fixed it.

**Comment by bcb12** [comment]
> Hello, per "Fix a crash launching 32 bit Vulkan titles (including DXVK SteamPlay)" in the [2022-10-24](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3428954061821341725) Steam client beta update, please opt back into the Steam client beta if you opted out to workaround this issue, and/or re-enable Steam's shader pre-caching subcomponent if you disabled that as a workaround and retest this issue.

Still getting the same behaviour. Fallout 3, Fallout New Vegas and GTA IV not launching when Shader Pre-Caching is enabled.

EDIT: got the update after restarting the pc. Now everything works perfectly.

**Comment by Skiski** [comment]
I've opt into the beta and evrey game I've tested so far works, including GTAIV.
I think we can mark it as solved.

**Comment by StarterX4** [comment]
I've also found out a weird issue that with the latest Steam client beta, text typing doesn't work in TF2, i can type nowhere in the whole game (chat, console, search fields), though the binds do work, as well as pasting the copied text. Again, opting out of the client beta fixed the issue.

@Skiski No, we can't. It's still broken.

**Comment by kisak-valve** [comment]
Hello @StarterX4, please open separate issue reports for your follow up issues in the Source-1-Games issue tracker.

**Comment by spiffeeroo** [comment]
The 32-bit games are working again in the newest Steam Beta build for me (so far). 👍

**Comment by dowttie** [comment]
I initially had the same issue with DXVK 32 bit games crashing entirely and now they work, but another issue arose which is that the overlay (and controller support) doesn't work with those same games. The overlay works fine when using wined3d instead
Games I've tested which are affected; Resident Evil 0, Resident Evil 5, Life Is Strange [not remastered]

**Comment by kisak-valve** [comment]
Hello @dowttie, please open a new issue report in the [steam-for-linux](https://github.com/ValveSoftware/steam-for-linux/issues) issue tracker so that your follow up issue can be tracked properly.

**Comment by kisak-valve** [comment]
Closing as fixed.
